llvm/test/Transforms/LoopVectorize/preserve-dbg-loc.ll

```diff
@@ -0,0 +1,33 @@
+; RUN: opt < %s -passes=debugify,loop-vectorize -force-vector-interleave=1 -force-vector-width=4 -prefer-inloop-reductions -S | FileCheck %s -check-prefix DEBUGLOC
```

Comment:
Can we keep the debug loc tests together in the existing `llvm/test/Transforms/LoopVectorize/preserve-dbg-loc-and-loop-metadata.ll`?

Also should have a case for VPWidenIntrinsics here.

Reply:
Is `-prefer-inloop-reductions` needed?

Reply:
I think the new added `llvm/test/Transforms/LoopVectorize/RISCV/preserve-dbg-loc.ll` contains test case for `VPWidenIntrinsicRecipe` (vp.select).

It's hard to generate in-loop reduction (VPReductionRecipe) without  `-prefer-inloop-reductions`.

Perhaps rename this test case to `preserve-dbg-loc-reduction-inloop.ll`?

Reply:
> I think the new added llvm/test/Transforms/LoopVectorize/RISCV/preserve-dbg-loc.ll contains test case for VPWidenIntrinsicRecipe (vp.select).

Yes but we should have a target-indepdent test if possible.

>Perhaps rename this test case to preserve-dbg-loc-reduction-inloop.ll?

Sounds good to me, thanks

Reply:
Add test case for VPWidenIntrinsicRecipe in default target.